### DISCUSSION TITLE
Self-reply threads and filtering

### DISCUSSION THREAD

**Discussion by IBBoard** [CodedOre/NewCaw#71] [Design]
By default, NewCaw hides replies. Which makes sense (particularly on profile views). I don't want to see everyone's reply to everyone else by default.

But it also excludes self-replies. So if someone posts a self-contained message (i.e. it [contains a full and coherent message](https://hachyderm.io/@ibboard/109767806196184057)) and replies to themselves with another self-contained post ([like this](https://hachyderm.io/@ibboard/109767963661722980)) then you don't see the full thread.

There is the line on the left-hand side. Is that our only indicator? I'm trying to work out what situations it appears in and what situations it doesn't (aside: at the moment, it looks like it incorrectly links it to the post just below!)

Maybe this is a Mastodon design choice that we're mirroring. But it feels odd compared to Twitter showing you the full self-reply thread.

**Comment by CodedOre** [comment]
I think this is `CollectionView` not properly displaying the posts.
The idea was to hide replies that stand alone, but show them when the post above them is their parent post.

As far as I'm concerned, this is a bug, so I make an issue from this discussion.

**Reply by IBBoard** [reply]
I was just adjusting the filters and found that it's something wrong on the initial load.

Post a message. Post a reply. Check your feed (you may need to restart - it might only happen on first open). It should be showing the first message and the line, but not the second one.

Enable "Replies", the post appears.

Disable "Replies", the post _does not_ disappear. It now shows the "correct" behaviour of showing self-replies regardless of the "replies" filter status.

**Reply by CodedOre** [reply]
Yes, something there has completely broke at some point.

Without having looked much into it, my guess it that `CollectionView` is failing to properly detect which posts to hide or show, as it is (as of now) responsible for this task.
